### PR TITLE
Feature/backend dev

### PR DESCRIPTION
**Pull request by dlobi**

### CODE REVIEW

File: backend/game/core.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def __init__(self, width=config.WIDTH, height=config.HEIGHT,
+        is_cyclic=config.IS_CYCLIC):
+        self.__dict__ = self.__shared_state
+        self.width = width
+        self.height = height
+        self.socket_to_player = dict()
+        self.is_cyclic = is_cyclic
+    
+    def create_new_player(self, websocket, name="Deuce"):
```
Comment (yuvallevy) [inline backend/game/core.py]: Note for later - check whether there is no player at the chosen position

File: backend/game/core.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class Game(object):
+    """
+    Core game singleton. Holds game data and dispatches it to clients
+    """
+    __shared_state = {}
+
+    def __init__(self, width=config.WIDTH, height=config.HEIGHT,
+        is_cyclic=config.IS_CYCLIC):
+        self.__dict__ = self.__shared_state
```
Comment (kfiri) [inline backend/game/core.py]: למה לעשות את זה אם גם ככה אתה יכול לגשת למשתנה הזה דרך self?

File: backend/game/player.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from uuid import uuid4
+
+class Player(object):
+
+    def __init__(self, websocket, x=0, y=0, name="Deuce", score=0):
```
Comment (kfiri) [inline backend/game/player.py]: name="Deuce"? why does this has a default value?

File: backend/game/core.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        for player_socket in self.socket_to_player.keys():
+            player_socket.write_message(json.loads({
+                'type': 'info',
+                'data': game_state
+            }))
+
+    def send_knockack(self, player_socket):
+        player_socket.write_message(json.loads({
+            'type': 'knockack',
```
Comment (kfiri) [inline backend/game/core.py]: We should use constants for these

File: backend/game/client_handler.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import logbook
+from tornado.websocket import WebSocketHandler
+
+from .core import Game
+from .. import config
+
+class ClientWebSocketHandler(WebSocketHandler):
+
+    game = Game()
+    message_handlers = dict()
```
Comment (kfiri) [inline backend/game/client_handler.py]: This should have a more indicative name in order to understand the meaning of this variable

**Review by kfiri — CHANGES_REQUESTED** [review]
I <3 U

File: backend/game/core.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+from .player import Player
+from . import message
+from .. import config
+
+class Game(object):
+    """
+    Core game singleton. Holds game data and dispatches it to clients
+    """
+    __instance = None
```
Comment (kfiri) [inline backend/game/core.py:12]: למה לא magic?

**Pull request merged**